File: backend/app/schemas/resume.py
from pydantic import BaseModel
from typing import List, Optional


class PersonalInfo(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    linkedin: Optional[str] = None
    github: Optional[str] = None
    portfolio: Optional[str] = None


class Education(BaseModel):
    degree: Optional[str] = None
    institution: Optional[str] = None
    start_year: Optional[str] = None
    end_year: Optional[str] = None
    grade: Optional[str] = None


class Experience(BaseModel):
    job_title: Optional[str] = None
    company: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    description: List[str] = []


class Project(BaseModel):
    name: Optional[str] = None
    technologies: List[str] = []
    description: List[str] = []


class ResumeAIData(BaseModel):
    personal_info: PersonalInfo
    summary: Optional[str] = None

    technical_skills: List[str] = []
    tools: List[str] = []
    concepts: List[str] = []

    education: List[Education] = []
    experience: List[Experience] = []
    projects: List[Project] = []

    certifications: List[str] = []
    achievements: List[str] = []

File: frontend/src/types/resume.ts
export type PersonalInfo = {
    name: string | null;
    email: string | null;
    phone: string | null;
    linkedin: string | null;
    github: string | null;
    portfolio: string | null;
};

export type Education = {
    degree: string | null;
    institution: string | null;
    start_year: string | null;
    end_year: string | null;
    grade: string | null;
};

export type Experience = {
    company?: string | null;
    role?: string | null;
    start_year?: string | null;
    end_year?: string | null;
    description?: string[];
};

export type Project = {
    name: string;
    technologies: string[];
    description: string[];
};

export type ResumeAIData = {
    personal_info: PersonalInfo;
    summary: string | null;

    technical_skills: string[];
    tools: string[];
    concepts: string[];

    education: Education[];
    experience: Experience[];
    projects: Project[];

    certifications: string[];
    achievements: string[];
};

export type Resume = {
    id: number;
    user_id: number;
    original_filename: string;
    file_path: string;
    extracted_text: string;
    ai_analysis: ResumeAIData | null;
    status: string;
    created_at: string;
    updated_at: string;
};